signer/xpi/cose.go

```diff
@@ -49,9 +49,19 @@ func (s *XPISigner) generateCOSEKeyPair(coseAlg *cose.Algorithm) (eeKey crypto.P
 
 	switch coseAlg {
 	case nil:
-		err = fmt.Errorf("Cannot generate private key for nil cose Algorithm")
+		err = errors.New("Cannot generate private key for nil cose Algorithm")
 	case cose.PS256:
-		size := s.issuerKey.(*rsa.PrivateKey).N.BitLen()
+		if s.issuerKey == nil {
+			err = errors.New("Cannot generate COSE key pair with nil issuerKey")
+			return
+		}
+		var size int
+		issuerRSAKey, ok := s.issuerKey.(*rsa.PrivateKey)
```

Comment:
wait, no, that's not what I meant. I meant that s.issuerKey could be an `*ecdsa.PrivateKey`, which would make this whole code block fail

Reply:
Wouldn't this fall back to the minKeySize?

I thought this test would cover that https://github.com/mozilla-services/autograph/blob/979c9c790375365563910b99d2d1e56944b389a0/signer/xpi/cose_test.go#L60-L71

Reply:
d'oh yeah, fixed to handle ecdsa and other types in https://github.com/mozilla-services/autograph/pull/76/commits/9bc34836847144f80737749520f425f0bc3facb2